lib/cylc/task_pool.py

```diff
@@ -1238,27 +1252,32 @@ def process_queued_db_ops(self):
                         self.pub_dao.add_update_item(
                             table_name, set_args, where_args)
 
-        # record any broadcast settings to be dumped out
-        bcast = BroadcastServer.get_inst()
-        for table_name, db_inserts in sorted(bcast.db_inserts_map.items()):
-            while db_inserts:
-                db_insert = db_inserts.pop(0)
-                self.pri_dao.add_insert_item(table_name, db_insert)
-                self.pub_dao.add_insert_item(table_name, db_insert)
-        for table_name, db_deletes in sorted(bcast.db_deletes_map.items()):
-            while db_deletes:
-                where_args = db_deletes.pop(0)
-                self.pri_dao.add_delete_item(table_name, where_args)
-                self.pub_dao.add_delete_item(table_name, where_args)
+        # Record suite parameters and tasks in pool
+        # Record any broadcast settings to be dumped out
+        for obj in self, BroadcastServer.get_inst():
```

Comment:
Running my 100-task busy suite for 30 cycles in my environment, `process_queued_db_ops` has a cumulative run time of 67.2s. This is faster when compared to master's 63.6s + 10.8s on state file dumping.